cmd/collector/app/span_handler_builder.go

```diff
@@ -76,7 +76,7 @@ func (b *SpanHandlerBuilder) BuildHandlers(spanProcessor processor.SpanProcessor
 			zs.NewChainedSanitizer(zs.NewStandardSanitizers()...),
 		),
 		handler.NewJaegerSpanHandler(b.Logger, spanProcessor),
-		handler.NewGRPCHandler(b.Logger, spanProcessor, tenancy.NewTenancyConfig(&b.CollectorOpts.GRPC.Tenancy)),
+		handler.NewGRPCHandler(b.Logger, spanProcessor, tenancy.NewTenancyManager(&b.CollectorOpts.GRPC.Tenancy)),
```

Comment:
we should treat TM similar to Logger - pass it to SpanHandlerBuilder at construction.

another peculiar thing: why is it `b.CollectorOpts.GRPC.Tenancy`? I.e. why tenancy options are within GRPC options?

Reply:
This line only became part of the PR because of the rename from `TenancyConfig` to `TenancyManager`.  The collector doesn't use the interceptors, which were introduced for query.  (I tend to use interceptors as a last resort, as many devs struggle with them.)  Now that I introduced them for query it makes sense to use them for collection as well, and I will refactor this code to use them.